src/include/btree_cmp_inline.h

```diff
@@ -31,70 +205,30 @@
 static WT_INLINE int
 __wt_lex_compare(const WT_ITEM *user_item, const WT_ITEM *tree_item)
 {
-    size_t len, usz, tsz;
-    const uint8_t *userp, *treep;
+    size_t len, tsz, usz;
+    int lencmp, ret_val;
 
     usz = user_item->size;
     tsz = tree_item->size;
-    len = WT_MIN(usz, tsz);
-
-    userp = (const uint8_t *)user_item->data;
-    treep = (const uint8_t *)tree_item->data;
-
-#ifdef HAVE_X86INTRIN_H
-    /* Use vector instructions if we'll execute at least 2 of them. */
-    if (len >= WT_VECTOR_SIZE * 2) {
-        size_t remain;
-        __m128i res_eq, u, t;
-
-        remain = len % WT_VECTOR_SIZE;
-        len -= remain;
-        if (WT_ALIGNED_16(userp) && WT_ALIGNED_16(treep))
-            for (; len > 0;
-                 len -= WT_VECTOR_SIZE, userp += WT_VECTOR_SIZE, treep += WT_VECTOR_SIZE) {
-                u = _mm_load_si128((const __m128i *)userp);
-                t = _mm_load_si128((const __m128i *)treep);
-                res_eq = _mm_cmpeq_epi8(u, t);
-                if (_mm_movemask_epi8(res_eq) != 65535)
-                    break;
-            }
-        else
-            for (; len > 0;
-                 len -= WT_VECTOR_SIZE, userp += WT_VECTOR_SIZE, treep += WT_VECTOR_SIZE) {
-                u = _mm_loadu_si128((const __m128i *)userp);
-                t = _mm_loadu_si128((const __m128i *)treep);
-                res_eq = _mm_cmpeq_epi8(u, t);
-                if (_mm_movemask_epi8(res_eq) != 65535)
-                    break;
-            }
-        len += remain;
-    }
-#elif defined(HAVE_ARM_NEON_INTRIN_H)
-    /* Use vector instructions if we'll execute at least 1 of them. */
-    if (len >= WT_VECTOR_SIZE) {
-        size_t remain;
-        uint8x16_t res_eq, u, t;
-        remain = len % WT_VECTOR_SIZE;
-        len -= remain;
-        for (; len > 0; len -= WT_VECTOR_SIZE, userp += WT_VECTOR_SIZE, treep += WT_VECTOR_SIZE) {
-            u = vld1q_u8(userp);
-            t = vld1q_u8(treep);
-            res_eq = vceqq_u8(u, t);
-            if (vminvq_u8(res_eq) != 255)
-                break;
-        }
-        len += remain;
+    if (usz < tsz) {
+        len = usz;
+        lencmp = -1;
+    } else if (usz > tsz) {
+        len = tsz;
+        lencmp = 1;
+    } else {
+        len = usz;
+        lencmp = 0;
     }
-#endif
-    /*
-     * Use the non-vectorized version for the remaining bytes and for the small key sizes.
-     */
-    for (; len > 0; --len, ++userp, ++treep)
-        if (*userp != *treep)
-            return (*userp < *treep ? -1 : 1);
 
-    /* Contents are equal up to the smallest length. */
-    return ((usz == tsz) ? 0 : (usz < tsz) ? -1 : 1);
+    if (len > WT_VECTOR_SIZE)
```

Comment:
Yes, the initial implementation is using your version. I want to let 1 bytes to fall into the short case so I changed it. I don't see any difference.

Reply:
I just remembered that the reason I sent 16 to the long path was that it allowed using `if (len & 8)` rather than `if (len >= 8)` in the short path which saves an instruction, but is incorrect for exactly 16. Not a huge deal, but worth considering. OTOH, if the `WT_COMPARE_SHORT_MAXLEN` path is notably faster, then the win from doing that when the key is exactly 16 bytes may be more important.